dali/dfu/dfurun.cpp

```diff
@@ -1453,13 +1453,13 @@ class CDFUengine: public CInterface, implements IDFUengine
                         if (diffNameSrc.get()||diffNameDst.get())
                         {
                             Owned<IFileDescriptor> oldf;
-                            oldf.setown(queryDistributedFileDirectory().getFileDescriptor(diffNameSrc,foreigncopy?foreignuserdesc:userdesc,foreigncopy?foreigndalinode:NULL));
+                            oldf.setown(queryDistributedFileDirectory().getFileDescriptor(diffNameSrc,AccessMode::readRandom,foreigncopy?foreignuserdesc:userdesc,foreigncopy?foreigndalinode:NULL));
```

Comment:
I think actually readSequential.

Reply:
not sure, see previous comment/question. index writes are sequential,
but 'sequential' index reads, are probably mostly sequential, but non-leaf nodes are after leaf nodes, and will seek backwards and read levels right?

Reply:
I think you're right that it probably does - but it shouldn't.  It would be much better to start from the first leaf node (which I think is always record 1).  It would be worth checking (and optimizing).

Reply:
for diff yes, and for indexes without filters.
For indexes without filters, yes it should be able to avoid reading any of the non-leaf nodes, and read the leaf nodes that followed the header node (if I remember layout correctly).

I don't know how significant an optimization it is in general...
Sounds like it should be left as readRandom for now (?), and a separate JIRA opened to look at sequential non-filtered index reads being optimized ?